echopype/tests/echodata/test_echodata_structure.py

```diff
@@ -0,0 +1,305 @@
+
+    # make sure that the value of the attribute is the right type
+    assert isinstance(ed[grp_path].attrs[attr], typ)
+    assert isinstance(tree[grp_path].ds.attrs[attr], typ)
+
+    # drop the attribute so we can directly compare datasets
+    del ed[grp_path].attrs[attr]
+    del tree[grp_path].ds.attrs[attr]
+
+
+def compare_ed_against_tree(ed, tree):
+    """
+    This function compares the Datasets
+    of ed against tree and makes sure they
+    are identical.
+
+    Parameters
+    ----------
+    ed : EchoData
+        EchoData object
+    tree : Datatree
+        Datatree object
+
+    Notes
+    -----
+    The Datatree object is created from an EchoData
+    object written to a netcdf file.
+    """
+
+    for grp_path in ed.group_paths:
+        if grp_path == "Top-level":
+            assert tree.ds.identical(ed[grp_path])
+        else:
+            assert tree[grp_path].ds.identical(ed[grp_path])
+
+
+def _get_conversion_file_lists(azfp_path, ek60_path, ek80_path):
+
+    converted_raw_paths_v06x = [ek60_path / "ek60-Summer2017-D20170615-T190214-ep-v06x.nc",
+                                ek60_path / "ek60-combined-ep-v06x.nc",
+                                ek80_path / "ek80-Summer2018--D20180905-T033113-ep-v06x.nc",
+                                ek80_path / "ek80-2018115-D20181213-T094600-ep-v06x.nc",
+                                ek80_path / "ek80-2019118-group2survey-D20191214-T081342-ep-v06x.nc",
+                                ek80_path / "ek80-Green2-Survey2-FM-short-slow-D20191004-T211557-ep-v06x.nc",
+                                azfp_path / "azfp-17082117_01A_17041823_XML-ep-v06x.nc"]
+
+    converted_raw_paths_v05x = [ek60_path / "ek60-Summer2017-D20170615-T190214-ep-v05x.nc",
+                                ek60_path / "ek60-combined-ep-v05x.nc",
+                                ek80_path / "ek80-Summer2018--D20180905-T033113-ep-v05x.nc",
+                                ek80_path / "ek80-2018115-D20181213-T094600-ep-v05x.nc",
+                                ek80_path / "ek80-2019118-group2survey-D20191214-T081342-ep-v05x.nc",
+                                ek80_path / "ek80-Green2-Survey2-FM-short-slow-D20191004-T211557-ep-v05x.nc",
+                                azfp_path / "azfp-17082117_01A_17041823_XML-ep-v05x.nc"]
+
+    return converted_raw_paths_v06x, converted_raw_paths_v05x
+
+
+def test_v05x_v06x_conversion_structure(azfp_path, ek60_path, ek80_path):
+    """
+    Tests that version 0.5.x echopype files
+    have been correctly converted to the
+    0.6.x structure.
+    """
+
+    converted_raw_paths_v06x, converted_raw_paths_v05x = \
+        _get_conversion_file_lists(azfp_path, ek60_path, ek80_path)
+
+    for path_v05x, path_v06x in zip(converted_raw_paths_v05x, converted_raw_paths_v06x):
+
+        print(f"path_v05x = {path_v05x}")
+
+        ed_v05x = open_converted(path_v05x)
+        tree_v06x = _tree_from_file(converted_raw_path=path_v06x)
```

Comment:
What's the reason for opening the 0.6.x files via datatree rather than via `open_converted`? As far as I can see, the results should be the same. But it looks like using `open_converted` would make `_tree_from_file` unnecessary and would make for a slightly better test (ie, actually opening a 0.6.x file via echopype's `open_converted` machinery).

Reply:
The reason I did not use `open_converted` to open the 0.6.x file is because I wanted the Datatree created to be as independent from the `EchoData` object as possible. I do this just incase we happen to make modifications to `EchoData` in the future that would somehow change the tree structure. I am not sure if this is absolutely necessary, but I do not see the harm in having it either.

Reply:
Thanks. Certainly no harm.